r/tests/testthat/test-Array.R

```diff
@@ -1172,6 +1172,43 @@ test_that("as_arrow_array() default method errors", {
   )
 })
 
+test_that("as_arrow_array() works for blob::blob()", {
+  skip_if_not_installed("blob")
+
+  expect_equal(
+    as_arrow_array(blob::blob(as.raw(1:5))),
```

Comment:
Should we test with more than length-1 lists? How about missing data? Any other corners of the `blob` implementation we need to be concerned with?

Should we assert that you actually get an Array?

Reply:
We could also test `blob(NULL)`, it is the NA value for BLOBs.